tests/integration.rs

```diff
@@ -7243,12 +7279,15 @@ mod dbus_api {
             guest.ssh_command("sync").unwrap();
             guest.ssh_command("sudo shutdown -H now").unwrap();
 
-            // Wait for the guest to be fully shutdown
-            thread::sleep(std::time::Duration::new(20, 0));
-
             // Then shutdown the VM
             assert!(dbus_api.remote_command("shutdown", None));
 
+            let shutdown_event = MetaEvent {
+                event: "shutdown".to_string(),
+                device_id: None,
+            };
+            assert!(wait_for_event(&shutdown_event, 20, &event_path));
+
```

Comment:
This is not right. You are asserting if  event `shtudown` is available before waiting for the same event... Then what's the purpose of using `wait_for_event()`?

Reply:
The problem is that `shutdown` event is not in events file before `dbus_api.remote_command("shutdown", None)`. I refactored it to run wait_for_event in a separate thread.

Reply:
What you are doing is incorrect. The `wait_for_event` you added does not achieve what the original `sleep()` was used for. Please read the test again, particularly: 

```
            // Sync and shutdown without powering off to prevent filesystem
            // corruption.
```

Reply:
`wait_for_event` is now implemented to sleep at least for 20s as original sleep() or until `shutdown` event is received. What is else is missing in `wait_for_event`?

Reply:
Let's try to understand the problem from a different angle. Think about why the original `thread::sleep(std::time::Duration::new(20, 0));` is put *before* the `assert!(dbus_api.remote_command("shutdown", None));` not *after*. To be clear, they will achieve different behavior. Also, think about what behavior that adding the `wait_for_event` will achieve. We can sync off line if you need more detailed explanation.